docs/policy.json.md

```diff
@@ -0,0 +1,248 @@
+<!-- NOTE: Keep this in sync with signature/policy_types.go! -->
+
+# Signature verification policy file format
+
+Signature verification policy files are used to specify policy, e.g. trusted keys,
+applicable when deciding whether to accept an image, or individual signatures of that image, as valid.
+
+The default policy is stored (unless overridden at compile-time) at `/etc/atomic/policy.json`;
```

Comment:
On Wed, Aug 10, 2016 at 05:11:20PM -0700, Miloslav Trmač wrote:
> The basedir spec is essentially the same hard-coded path as above, +
> extra environment variable overrides; but it still requires the
> hard-coded path to be defined (and in typical use of the basedir
> spec, the environment variables are unset and the hard-coded path
> applies).

The basedir spec defines a default list for configs:

  Lookups of the configuration file should search for
  ./subdir/filename relative to all base directories indicated by
  $XDG_CONFIG_HOME and $XDG_CONFIG_DIRS . If an environment variable
  is either not set or empty, its default value as defined by this
  specification should be used instead.

So using that approach means you have to define the relative portion
where you'll lookup your configuration
(e.g. ./containers/fetch-policy.d/… [1]), but you can leave the
absolute portion (e.g. /etc/xdg, $HOME/.config, …) up to the basedir
spec (and override it via environment variables, if you like).

> Still, even deferring to a third-party standard or configuration
> management API does not automatically resolve the question whether
> the signature verification policy “belongs” to the application or to
> the library — and my primary concern here is for all applications to
> use the same policy, i.e. for this repo to hard-code the location of
> the configuration within the configuration management API

I'd have it belong to the library, just like /etc/krb5.conf and
KRB5_CONFIG belong to mit-krb5 (and contain libdefaults).  I'd just
rather use the basedir spec instead of declaring a new
CONTAINER_CONFIG or hard-coding the path with no ability to override
at runtime.

[1]: https://github.com/containers/image/pull/50/files/26c93df48f7cc0f1148f6b63bd3680609bc2f47f#r74358446